prow/plugins/lgtm/lgtm.go

```diff
@@ -83,82 +96,108 @@ func handle(gc githubClient, log *logrus.Entry, e *github.GenericCommentEvent) e
 	// If we create an "/lgtm" comment, add lgtm if necessary.
 	// If we create a "/lgtm cancel" comment, remove lgtm if necessary.
 	wantLGTM := false
-	if lgtmRe.MatchString(e.Body) {
+	if lgtmRe.MatchString(body) {
 		wantLGTM = true
-	} else if lgtmCancelRe.MatchString(e.Body) {
+	} else if lgtmCancelRe.MatchString(body) {
 		wantLGTM = false
 	} else {
 		return nil
 	}
 
-	org := e.Repo.Owner.Login
-	repo := e.Repo.Name
-	commentAuthor := e.User.Login
+	// Author cannot LGTM own PR
+	isAuthor := author == issueAuthor
+	if isAuthor && wantLGTM {
+		resp := "you cannot LGTM your own PR."
+		log.Infof("Commenting with \"%s\".", resp)
+		return gc.CreateComment(repo.Owner.Login, repo.Name, number, plugins.FormatResponseRaw(body, htmlURL, author, resp))
+	}
+
+	return handle(wantLGTM, author, issueAuthor, repo, assignees, number, body, htmlURL, gc, log)
+}
 
-	// Allow authors to cancel LGTM. Do not allow authors to LGTM, and do not
-	// accept commands from any other user.
+func handlePullRequestReview(gc githubClient, log *logrus.Entry, e github.ReviewEvent) error {
```

Comment:
This still needs to by gated, as it shouldn't be the default: https://github.com/kubernetes/test-infra/pull/6738#issuecomment-378332193

Reply:
I looked at your options add on for `/approve` (#7504) and I can't tell how those would translate to here.
I understand the option for `lgtm` to act as `approve` already exists in the plugin config.
It is unclear on what logic is missing where in this file.

Reply:
Okay, I see that this is actually a separate flag from `LgtmActsAsApprove`.

I'm still really confused on where to add this. :/

Reply:
Okay, so I see you've created `ReviewActsAsLgtm`, but you've added it as an option to the Approve plugin struct. You'd want to create one for the LGTM plugin.

Something like this pulls the options from the config:
https://github.com/kubernetes/test-infra/blob/ef15bf16785fb3bc087cea841ec3723b1b652efc/prow/plugins/approve/approve.go#L453-L464

And something like this pulls it from the config and into a struct:
https://github.com/kubernetes/test-infra/blob/ef15bf16785fb3bc087cea841ec3723b1b652efc/prow/plugins/approve/approve.go#L137

Then you could compare `opts.ReviewActsAsLgtm` as a bool.